### PR TITLE
refactored error handling of mongo errors in directory api, fixed #55

### LINKED ISSUE #55: Don't return mongo errors to the client
`404 Client Error: mongo: no documents in result for url: https://explorer.testnet.grid.tf/explorer/farms/1`

### PR DESCRIPTION
**Pull request by DylanVerstraete**

### CODE REVIEW

File: pkg/directory/farms_handlers.go [hunk 1]
```diff
@@ -34,7 +40,7 @@ func (s *FarmAPI) registerFarm(r *http.Request) (interface{}, mw.Response) {
 
 	id, err := s.Add(r.Context(), db, info)
 	if err != nil {
-		return nil, mw.Error(err)
+		return nil, mw.GenericDBError(mw.DBError{Cause: err, Message: errFailedToRegisterFarm.Error()})
```
Comment (muhamadazmy) [inline pkg/directory/farms_handlers.go:43]: can't we have a shortcut method for this ? your input is just the source error, and a message. 
May be something like
```go
mw.GenericDBError(cause, msg) 
```
or even 
```go
mw.GenericDBError(Case, msg, code...) 
```

Comment (DylanVerstraete) [reply pkg/directory/farms_handlers.go:43]: Yes indeed

File: mw/action.go [hunk 2]
```diff
@@ -111,6 +113,20 @@ func Error(err error, code ...int) Response {
 	return genericResponse{status: status, err: err}
 }
 
+// GenericDBError generic db error response
+func GenericDBError(err error, message string, code ...int) Response {
```
[the remark this answers is not in the document]
Comment (DylanVerstraete) [reply mw/action.go:117]: I deleted that on accident

File: pkg/directory/farms_handlers.go [hunk 3]
```diff
@@ -56,7 +62,7 @@ func (s *FarmAPI) updateFarm(r *http.Request) (interface{}, mw.Response) {
 
 	farm, err := s.GetByID(r.Context(), db, id)
 	if err != nil {
-		return nil, mw.NotFound(err)
+		return nil, mw.GenericDBError(err, fmt.Sprintf("farm with id %s not found", sid))
```
[the remark this answers is not in the document]
Comment (DylanVerstraete) [reply pkg/directory/farms_handlers.go]: Refactored this as well.

File: mw/action.go [hunk 4]
```diff
@@ -111,6 +113,20 @@ func Error(err error, code ...int) Response {
...
+	status := http.StatusInternalServerError
+	if len(code) > 0 {
+		status = code[0]
+	}
+
+	if errors.Is(err, mongo.ErrNoDocuments) {
```
[the remark this answers is not in the document]
Comment (DylanVerstraete) [reply mw/action.go:123]: Is it not better to not expose any other errors from mongo?

**Comment by DylanVerstraete** [comment]
@zaibon So what are we going to do with this? What approach do you think we should take?

**Pull request closed without merging**